python/ray/experimental/channel/common.py

```diff
@@ -29,7 +33,41 @@ def register_custom_serializer(self) -> None:
         default device. Instead, these should be extracted from the
         worker-local _SerializationContext.
         """
-        pass
+        if self._contains_type is not None:
+            self._contains_type.register_custom_serializer()
+
+    @property
+    def is_direct_return(self) -> bool:
```

Comment:
is it expensive to figure this out automatically>? For example,

- if the return type is tensor (with a type hint tensor), it is True
- If not, we can set it False automatically
?

Reply:
Python doesn't guarantee that if it's true, it will always be true :( Also, when we use direct_return=True, the only thing we can do is loop on a NCCL channel. We might be able to figure out a way to send some control message along the NCCL channel, but it seems a bit complicated for now.